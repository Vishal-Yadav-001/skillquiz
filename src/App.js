import React, { useEffect, useState } from "react";
import { Navigate, Route, Routes, useNavigate } from "react-router-dom";
import Home from "./components/home/home.component";
import Questions from "./components/questionsTab/questions.component";
import Register from "./components/register/register.component.jsx";
import { topics, levels, limit } from "./components/constants/constant.jsx";
import axios from "axios";
import Scores from "./components/score/scores.component";
import { CssBaseline, ThemeProvider } from "@mui/material";
import Notifications from "./components/notifications/notifications.component";
import defaultTheme from "./components/theme/theme.component";
import NotFound from "./components/notfound.component";
function App() {
  const navbarHeight = "64px";
  const [questions, setQuestions] = useState([]);
  const [isLogined, setIsLogined] = useState(() =>
    localStorage.key(0) ? true : false
  );
  const [selectedTopic, setSelectedTopic] = useState("");
  const [questionLevel, setQuestionLevel] = useState("");
  const [questionsRange, setQuestionsRange] = useState("");
  const [userAgreed, setUserAgreed] = useState(false);
  const [mobileOpen, setMobileOpen] = useState(false);
  const [userResponse, setUserResponse] = useState(null);
  const [retry, setRetry] = useState(false);
  const [showNotification, setShowNotification] = useState(false);
  const [notification, setNotification] = useState("");
  const [isLoading, setIsLoading] = useState(true);
  const [errorOccurred, setErrorOccurred] = useState(false);
  const [showReturnDialog, setShowReturnDialog] = useState(false);
  const [previousUserResponse, setPreviousUserResponse] = useState(null);
  const user = JSON.parse(localStorage.getItem("User"));
  const navigate = useNavigate();
  useEffect(() => {
    if (user) {
      const { accidentalClose } = user;
      if (accidentalClose) {
        setShowReturnDialog(true);
      }
    }
  }, []);

  useEffect(() => {
    const getQuestions = async () => {
      try {
        if (user && userAgreed) {
          const API_KEY = "V7RxLSLo3E2DHXbKsRe3e6PLFsvCtlOg2GI8lJSh";
          if (!retry) {
            const response = await axios.get(
              `https://quizapi.io/api/v1/questions?apiKey=${API_KEY}&tags=${selectedTopic}&difficulty=${questionLevel}&limit=${questionsRange}`
            );
            if (response.status) {
              setQuestions(() => constructObject(response.data));

              setIsLoading(false);
              setErrorOccurred(false);
            }
          }
          if (retry) {
            const userData = JSON.parse(localStorage.getItem("User"));
            if (userData) {
              setQuestions(() => userData?.providedQuestions);
              if (userData?.accidentalClose) {
                const { userResponse, selectedTopic, providedQuestionsLevel } =
                  userData;
                setPreviousUserResponse(() => {
                  return { ...userResponse };
                });
                setSelectedTopic(() => selectedTopic);
                setQuestionLevel(() => providedQuestionsLevel);
              }
              setIsLoading( false);
              setErrorOccurred(false);
            }
          }
        }
      } catch (error) {
        setNotification( { message: error.message, type: "error" }
        );
        setShowNotification(true);
        setErrorOccurred( true);
        setIsLoading(false);
      }
    };
    getQuestions();

    return () => setShowNotification(false);
  }, [userAgreed]);
  /**
   * Construct questions array to be used for displaying quiz questions.
   * @param {*} data - response from api
   * @returns array of object - containing questions,options,answers
   */
  const constructObject = (data) => {
    let questionsList = [];
    data?.map((list, index) => {
      questionsList.push({
        id: index + 1,
        question: list?.question,
        options: constructOptions(list?.answers),
        answer: constructAnswer(list?.correct_answers),
      });
    });
    return questionsList;
  };
  /**
   * Construct array from Object
   * @param {*} listOfOptions - Object containing options for questions 
   * @returns array of options.
   */
  const constructOptions = (listOfOptions) => {
    const optionsArray = [];
    for (const key in listOfOptions) {
      if (listOfOptions[key]) {
        const optionsObject = {
          id: optionsArray.length + 1,
          value: listOfOptions[key],
        };
        optionsArray.push(optionsObject);
      }
    }
    return optionsArray;
  };
  /**
   * Filter out correct answer
   * @param {*} answers - object containing answers
   * @returns Object containing correct answer
   */
  const constructAnswer = (answers) => {
    let response;

    const answerObj = [
      {
        id: 1,
        value: "answer_a_correct",
      },
      {
        id: 2,
        value: "answer_b_correct",
      },
      {
        id: 3,
        value: "answer_c_correct",
      },
      {
        id: 4,
        value: "answer_d_correct",
      },
    ];

    for (const key in answers) {
      if (answers[key] === "true") {
        response = answerObj.find((data) => data.value === key);
      }
    }
    return response;
  };

  const handleDrawerToggle = () => {
    setMobileOpen(!mobileOpen);
  };
  /**
   *  When is user has left quiz mid way, user wil be shown option to continue quiz or start fresh.
   * Resume -> redirect user to question page , start timer from the time at which user left quiz also display previous responses.
   */
  const handleResume = () => {
    setShowReturnDialog(false);
    setUserAgreed( true);
    setRetry(true);
    setUserResponse(null);
    setQuestionsRange(() => {
      return user?.provideQuestionsCount;
    });
    navigate("/questions");
  };
  /**
   *  Start Fresh -> Helps user to start quiz from start.
   */
  const handleStartFresh = () => {
    const userResponseObj = {
      ...user,
      accidentalClose: false,
    };
    localStorage.setItem("User", JSON.stringify(userResponseObj));
    setShowReturnDialog(() => false);
  };

  return (
    <ThemeProvider theme={defaultTheme}>
      <CssBaseline/>
      <div className="App">
        <Notifications
          notification={notification}
          showNotification={showNotification}
          setShowNotification={setShowNotification}
        />
        <Routes>
          <Route
            path="/"
            data-testid="Home"
            element={
              <Home
                navbarHeight={navbarHeight}
                questions={questions}
                setQuestions={setQuestions}
                setIsLogined={setIsLogined}
                isLogined={isLogined}
                selectedTopic={selectedTopic}
                setSelectedTopic={setSelectedTopic}
                questionLevel={questionLevel}
                setQuestionLevel={setQuestionLevel}
                questionsRange={questionsRange}
                setQuestionsRange={setQuestionsRange}
                topics={topics}
                levels={levels}
                limit={limit}
                userAgreed={userAgreed}
                setUserAgreed={setUserAgreed}
                user={user}
                constructObject={constructObject}
                handleDrawerToggle={handleDrawerToggle}
                setShowNotification={setShowNotification}
                setNotification={setNotification}
                showReturnDialog={showReturnDialog}
                handleResume={handleResume}
                handleStartFresh={handleStartFresh}
              />
            }
          >
            <Route
              path="/questions"
              element={
                <Questions
                  questions={questions}
                  selectedTopic={selectedTopic}
                  questionLevel={questionLevel}
                  questionsRange={questionsRange}
                  navbarHeight={navbarHeight}
                  mobileOpen={mobileOpen}
                  setMobileOpen={setMobileOpen}
                  handleDrawerToggle={handleDrawerToggle}
                  userResponse={userResponse}
                  setUserResponse={setUserResponse}
                  setUserAgreed={setUserAgreed}
                  isLoading={isLoading}
                  setIsLoading={setIsLoading}
                  errorOccurred={errorOccurred}
                  setErrorOccurred={setErrorOccurred}
                  previousUserResponse={previousUserResponse}
                  setPreviousUserResponse={setPreviousUserResponse}
                  setIsLogined={setIsLogined}
                  setQuestions={setQuestions}
                />
              }
            ></Route>
          </Route>
          <Route
            path="score"
            element={
              <Scores
                setUserAgreed={setUserAgreed}
                setRetry={setRetry}
                setUserResponse={setUserResponse}
                setQuestions={setQuestions}
              />
            }
          ></Route>
          <Route
            path="/register"
            element={
              <Register
                setIsLogined={setIsLogined}
                setShowNotification={setShowNotification}
                setNotification={setNotification}
                setUserAgreed={setUserAgreed}
              />
            }
          />
          <Route path="*" element={<NotFound/>} />
        </Routes>
      </div>
    </ThemeProvider>
  );
}

export default App;
