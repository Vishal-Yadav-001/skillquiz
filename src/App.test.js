import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { rest } from 'msw'
import { setupServer } from 'msw/node'
import App from "./App";

describe("<App />", () => {
  const server = setupServer(
    rest.get(
      "https://quizapi.io/api/v1/questions?apiKey=V7RxLSLo3E2DHXbKsRe3e6PLFsvCtlOg2GI8lJSh&tags=HTML&difficulty=Easy&limit=5",
      (req, res, ctx) => res(ctx.json(apiResponse))
    )
  );

  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  test("show attention popup when user is bac after not finishing quiz ", () => {
    localStorage.setItem("User", JSON.stringify(userData));
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(screen.getByText("Attention!")).toBeInTheDocument();
  });

  test("snapshot App", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("call api for response ", () => {
    localStorage.setItem("User", JSON.stringify(userData));
   const app =  render(
      <BrowserRouter>
        <App userAgreed={true} />
      </BrowserRouter>
    );
    console.log(app)
  
  });
});

const userData = {
  name: "xvxvxxv",
  email: "xvxvxcv@gmail.com",
  password: "vxvxcv",
  selectedTopic: "HTML",
  providedQuestions: [
    {
      id: 1,
      question: "What is the correct HTML for making a text area?",
      options: [
        { id: 1, value: '<input type="textbox">' },
        { id: 2, value: "<textarea>" },
        { id: 3, value: '<input type="textarea">' },
        { id: 4, value: "<textbox>" },
      ],
      answer: { id: 2, value: "answer_b_correct" },
    },
    {
      id: 2,
      question: "Which tag creates a check box for a form in HTML?",
      options: [
        { id: 1, value: "<checkbox>" },
        { id: 2, value: "<input checkbox>" },
        { id: 3, value: "<input=checkbox>" },
        { id: 4, value: '<input type="checkbox">' },
      ],
      answer: { id: 4, value: "answer_d_correct" },
    },
    {
      id: 3,
      question: "HTML supports",
      options: [
        { id: 1, value: "ordered lists" },
        { id: 2, value: "unordered lists" },
        { id: 3, value: "both type of lists" },
        { id: 4, value: "does not support those types" },
      ],
      answer: { id: 3, value: "answer_c_correct" },
    },
    {
      id: 4,
      question: "How can you make A bulleted list with numbers?",
      options: [
        { id: 1, value: "<dl>" },
        { id: 2, value: "<ul>" },
        { id: 3, value: "<ol>" },
        { id: 4, value: "<list>" },
      ],
      answer: { id: 3, value: "answer_c_correct" },
    },
    {
      id: 5,
      question:
        "All elements are identified by their __________ and are marked up using either start tags and end tags or self-closing tags",
      options: [
        { id: 1, value: "Attribute Names" },
        { id: 2, value: "Tag Names" },
        { id: 3, value: "Class Names" },
        { id: 4, value: "None of the mentioned" },
      ],
      answer: { id: 2, value: "answer_b_correct" },
    },
  ],
  provideQuestionsCount: 5,
  providedQuestionsLevel: "Easy",
  totalTimeProvided: 100,
  timeSpent: 44,
  userResponse: {
    1: { id: 2, value: "<textarea>" },
    2: { id: 2, value: "<input checkbox>" },
    3: { id: 2, value: "unordered lists" },
  },
  questionAttempted: 3,
  score: 1,
  allAttempted: false,
  accidentalClose: true,
};

const apiResponse = [
  {
    id: 527,
    question: "HTML supports",
    description: null,
    answers: {
      answer_a: "ordered lists",
      answer_b: "unordered lists",
      answer_c: "both type of lists",
      answer_d: "does not support those types",
      answer_e: null,
      answer_f: null,
    },
    multiple_correct_answers: "false",
    correct_answers: {
      answer_a_correct: "false",
      answer_b_correct: "false",
      answer_c_correct: "true",
      answer_d_correct: "false",
      answer_e_correct: "false",
      answer_f_correct: "false",
    },
    correct_answer: "answer_a",
    explanation: null,
    tip: null,
    tags: [{ name: "HTML" }],
    category: "Linux",
    difficulty: "Easy",
  },
  {
    id: 476,
    question: "What should be the first tag in any HTML Document?",
    description: null,
    answers: {
      answer_a: "<html>",
      answer_b: "<!doctype html>",
      answer_c: "<title>",
      answer_d: "<head>",
      answer_e: null,
      answer_f: null,
    },
    multiple_correct_answers: "false",
    correct_answers: {
      answer_a_correct: "false",
      answer_b_correct: "true",
      answer_c_correct: "false",
      answer_d_correct: "false",
      answer_e_correct: "false",
      answer_f_correct: "false",
    },
    correct_answer: "answer_a",
    explanation: null,
    tip: null,
    tags: [{ name: "HTML" }],
    category: "Code",
    difficulty: "Easy",
  },
  {
    id: 162,
    question:
      "Which HTML element is used to specify a header for a document or section?",
    description: null,
    answers: {
      answer_a: "<top>",
      answer_b: "<header>",
      answer_c: "<head>",
      answer_d: "<section>",
      answer_e: null,
      answer_f: null,
    },
    multiple_correct_answers: "false",
    correct_answers: {
      answer_a_correct: "false",
      answer_b_correct: "true",
      answer_c_correct: "false",
      answer_d_correct: "false",
      answer_e_correct: "false",
      answer_f_correct: "false",
    },
    correct_answer: "answer_b",
    explanation: null,
    tip: null,
    tags: [{ name: "HTML" }],
    category: "Linux",
    difficulty: "Easy",
  },
  {
    id: 533,
    question: "In HTML document the tags",
    description: null,
    answers: {
      answer_a: "Should be written in upper case",
      answer_b: "can be written in both uppercase or lowercase",
      answer_c: "should be written in propercase",
      answer_d: "should be written in lower case",
      answer_e: null,
      answer_f: null,
    },
    multiple_correct_answers: "false",
    correct_answers: {
      answer_a_correct: "false",
      answer_b_correct: "true",
      answer_c_correct: "false",
      answer_d_correct: "false",
      answer_e_correct: "false",
      answer_f_correct: "false",
    },
    correct_answer: "answer_a",
    explanation: null,
    tip: null,
    tags: [{ name: "HTML" }],
    category: "Code",
    difficulty: "Easy",
  },
  {
    id: 543,
    question: "What is cite tag in HTML5?",
    description: null,
    answers: {
      answer_a: '<cite title"value">Some Text Here</cite>',
      answer_b: '<cite title:"value">Some Text Here</cite>',
      answer_c: '<cite title="value">Some Text Here</cite>',
      answer_d: null,
      answer_e: null,
      answer_f: null,
    },
    multiple_correct_answers: "false",
    correct_answers: {
      answer_a_correct: "false",
      answer_b_correct: "false",
      answer_c_correct: "true",
      answer_d_correct: "false",
      answer_e_correct: "false",
      answer_f_correct: "false",
    },
    correct_answer: "answer_a",
    explanation: null,
    tip: null,
    tags: [{ name: "HTML" }],
    category: "Code",
    difficulty: "Easy",
  },
];
