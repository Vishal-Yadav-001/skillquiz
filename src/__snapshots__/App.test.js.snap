// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot App 1`] = `
<div
  className="App"
>
  <div />
  <div
    className="MuiBox-root css-lkqzy"
  >
    <div
      className="MuiBox-root css-i9gxme"
    >
      <header
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-qv9cjy-MuiPaper-root-MuiAppBar-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <button
            aria-label="menu"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </button>
          <div
            className="MuiTypography-root MuiTypography-h6 css-195gxv0-MuiTypography-root"
          >
            SkillScore
          </div>
          <div
            className="MuiAvatar-root MuiAvatar-circular css-itb7qy-MuiAvatar-root"
          >
            <img
              alt="User Name"
              className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="young-man.png"
            />
          </div>
          <p
            className="MuiTypography-root MuiTypography-body1 css-2lae6c-MuiTypography-root"
          >
            xvxvxxv
          </p>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-10vccbw-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Logout
          </button>
        </div>
      </header>
    </div>
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg css-cswa1n-MuiContainer-root"
    >
      <div
        className="MuiBox-root css-19midj6"
      >
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiBox-root css-jmdsto"
          >
            <h4
              className="MuiTypography-root MuiTypography-h4 css-1lw9fw4-MuiTypography-root"
            >
              Choose a topic
            </h4>
            <p
              className="MuiTypography-root MuiTypography-body1 css-u030s2-MuiTypography-root"
              varient="h6"
            >
              Test you knowledge on various topics
            </p>
          </div>
          <div
            className="MuiStack-root css-1fkxkcb-MuiStack-root"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1i5fr42-MuiPaper-root-MuiCard-root"
            >
              <div
                className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  className="MuiBox-root css-xi606m"
                >
                  <img
                    alt="HTML"
                    height="75px"
                    src="html-5.png"
                    width="auto"
                  />
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-i0r1ze-MuiTypography-root"
                  >
                    HTML
                  </p>
                  <div />
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-dm1joi-MuiButtonBase-root-MuiButton-root"
                    data-testid="takeQuizButton"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiTypography-root MuiTypography-body css-6dq5cv-MuiTypography-root"
                      onClick={[Function]}
                    >
                      Take a quiz
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1i5fr42-MuiPaper-root-MuiCard-root"
            >
              <div
                className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  className="MuiBox-root css-xi606m"
                >
                  <img
                    alt="JavaScript"
                    height="75px"
                    src="js.png"
                    width="auto"
                  />
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-i0r1ze-MuiTypography-root"
                  >
                    JavaScript
                  </p>
                  <div />
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-dm1joi-MuiButtonBase-root-MuiButton-root"
                    data-testid="takeQuizButton"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiTypography-root MuiTypography-body css-6dq5cv-MuiTypography-root"
                      onClick={[Function]}
                    >
                      Take a quiz
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1i5fr42-MuiPaper-root-MuiCard-root"
            >
              <div
                className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  className="MuiBox-root css-xi606m"
                >
                  <img
                    alt="Sql"
                    height="75px"
                    src="sql-server.png"
                    width="auto"
                  />
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-i0r1ze-MuiTypography-root"
                  >
                    Sql
                  </p>
                  <div />
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-dm1joi-MuiButtonBase-root-MuiButton-root"
                    data-testid="takeQuizButton"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiTypography-root MuiTypography-body css-6dq5cv-MuiTypography-root"
                      onClick={[Function]}
                    >
                      Take a quiz
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1i5fr42-MuiPaper-root-MuiCard-root"
            >
              <div
                className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  className="MuiBox-root css-xi606m"
                >
                  <img
                    alt="DevOps"
                    height="75px"
                    src="devops.png"
                    width="auto"
                  />
                  <p
                    className="MuiTypography-root MuiTypography-body1 css-i0r1ze-MuiTypography-root"
                  >
                    DevOps
                  </p>
                  <div />
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-dm1joi-MuiButtonBase-root-MuiButton-root"
                    data-testid="takeQuizButton"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiTypography-root MuiTypography-body css-6dq5cv-MuiTypography-root"
                      onClick={[Function]}
                    >
                      Take a quiz
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer MuiBox-root css-rtkb4j"
    >
      <span>
        Made in 2022 by Vishal Yadav
      </span>
    </div>
  </div>
</div>
`;
